### DISCUSSION TITLE
Custom request not in validation method

### DISCUSSION THREAD

**Discussion by lionalex** [dedoc/scramble#217] [Q&A]
How can I explain custom request that aren't in validation mathod but in custom request Laravel?
For example:

```
 public function signup(UserSignupRequest $request, User $model)
 {
      return $this->sendSuccess('Registrazione effettuata con successo', $data);
 }
```

UserSignupRequest is a custom Request where are the validation method. How can I explain what are the requested variable in this method?

**Comment by romalytvynenko** [comment]
Hey @lionalex 

Please show the source of `UserSignupRequest`, and show what is the resulting docs of this endpoint.

Scramble should handle that automatically.

**Comment by lionalex** [comment]
Ok thanks.
But if I have this function for signin

```
public function signin(Request $request)
{
        $this->validateLogin($request);
}
```

where "validateLogin" is an AuthenticatesUsers method, I have not any detail

![Screenshot 2023-09-04 alle 07 56 10](https://github.com/dedoc/scramble/assets/4151825/d8548b81-0b9a-4f46-ab8d-137cc5fe1d36)

How can add details to this scenario?

**Reply by romalytvynenko** [reply]
@lionalex show the entire `AuthenticatesUsers`, please

**Comment by lionalex** [comment]
```
trait AuthenticatesUsers
{
    use RedirectsUsers, ThrottlesLogins;

    .....

    protected function validateLogin(Request $request)
    {
        $request->validate([
            $this->username() => 'required|string',
            'password' => 'required|string',
        ]);
    }

    .....

}
```

**Comment by romalytvynenko** [comment]
@lionalex 

This code pattern is not analyzable for now. 

You can find analyzable patterns in DOCS: https://scramble.dedoc.co/usage/request
